src/hotspot/share/cds/metaspaceShared.cpp

```diff
@@ -813,13 +832,18 @@ void MetaspaceShared::preload_and_dump(TRAPS) {
     }
   }
 
-  if (!CDSConfig::old_cds_flags_used()) {
-    // The JLI launcher only recognizes the "old" -Xshare:dump flag.
-    // When the new -XX:AOTMode=create flag is used, we can't return
-    // to the JLI launcher, as the launcher will fail when trying to
-    // run the main class, which is not what we want.
-    tty->print_cr("AOTCache creation is complete: %s", AOTCache);
-    vm_exit(0);
+  if (CDSConfig::new_aot_flags_used()) {
+    if (CDSConfig::is_dumping_preimage_static_archive()) {
+      tty->print_cr("AOTConfiguration recorded: %s", AOTConfiguration);
+      vm_exit(0);
+    } else {
```

Comment:
Is it appropriate to add assert of `CDSConfig::is_dumping_final_static_archive()` in the `else` case?

Reply:
In the `else` case, we could be dumping either "final" or "classic" static archive. This function is invoked only when dumping static archives, so I think extra asserts aren't necessary here.